src/main/java/com/capstone/doctorsOffice/services/PatientServiceImpl.java

```diff
@@ -0,0 +1,57 @@
+package com.capstone.doctorsOffice.services;
+
+import com.capstone.doctorsOffice.dtos.PatientDto;
+import com.capstone.doctorsOffice.entities.Patient;
+import com.capstone.doctorsOffice.repositories.PatientRepository;
+import jakarta.transaction.Transactional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class PatientServiceImpl implements PatientService {
+
+    @Autowired
+    private PatientRepository patientRepository;
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
+
+    @Override
+    public List<String> addPatient(PatientDto patientDto) {
+        List<String> response = new ArrayList<>();
+        Patient patient = new Patient(patientDto);
+        patientRepository.saveAndFlush(patient);
+        response.add("http://localhost:8080/login.html");
+        return response;
+    }
+
+    @Override
+    @Transactional
+    public List<String> patientLogin(PatientDto patientDto){
+        List<String> response = new ArrayList<>();
+        Optional<Patient> patientOptional = patientRepository.findByName(patientDto.getName());
```

Comment:
There is no unique constraint on the patient name, so we do not want to lookup a patient by their name for login. There could be 10 John Smith in the database.

We should add an email field to the Patient Entity and add a unique constraint to it
```
@Column(nullable = false, unique = true)
private String email;
```

Then you can do the look up by the email
```
Optional<Patient> patientOptional = patientRepository.findByEmail(patientDto.getEmail());
```

Reply:
Ops you're right, don't know why I used Name.
Done